ObjectCreation/AeroFoil.FCMacro

```diff
@@ -0,0 +1,1747 @@
+NACA_5_2ND_3RD_DIGITS = ["10", "20", "30", "40", "50", "21", "31", "41", "51"]
+FUNCTIONSARRAY = [
+    "+",
+    "-",
+    "*",
+    "/",
+    "^",
+    "e",
+    "pi",
+    "ln(",
+    "log(",
+    "sqrt(",
+    "sin(",
+    "cos(",
+    "tan(",
+    "asin(",
+    "acos(",
+    "atan(",
+    "x",
+    "(",
+    ")",
+    ".",
+]
+UNITSCONVERSION = [0, 1, 10, 1000, 25.4, 304.8, 914.4]
+MIN_CHORD_LENGTH_MM = 1
+MD_ = App.getUserMacroDir(True) + "/AeroFoil_UI_Files/AeroFoil_"
+
+
+
+# Global Variables
+# ------------------------------------------------------------------------------------------------
+
+# Variables Pertaining to the Entire Program
+dialogIndex = [1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
+
+
+
+# Main Function Class
+# ------------------------------------------------------------------------------------------------
+
+class AeroFoil:
+    def __init__(self, obj):
+        global UNITSCONVERSION
+        self.profileName = generateName("AeroFoil")
+        self.chord = obj.chordLength * UNITSCONVERSION[obj.chordUnits]
+        self.designtype = [obj.workbench, obj.curveType]
+        self.designclosed = obj.closed_
+        self.points = []
+        self.points.append(obj.pointsX)
+        self.points.append(obj.pointsY)
+        self.n_units = len(self.points[0])
+        self.objRefData = [
+            obj.airfoilType,
+            obj.airfoil4DNumber,
+            obj.airfoil5DNumber,
+            obj.airfoilProfileType,
+            obj.importFrom,
+        ]
+        self.designprogressbar = obj.progressBar_
+
+    def copy(self, quantity_):
+        if quantity_ != 1:
+            refNum = int(re.findall(r"\d+", self.profileName)[0])
+            FreeCADGui.Selection.clearSelection()
+            FreeCADGui.Selection.addSelection(
+                App.ActiveDocument.getObjectsByLabel(self.profileName)[0]
+            )
+            FreeCADGui.runCommand("Std_Copy")
+            for i in range(quantity_ - 1):
+                FreeCADGui.runCommand("Std_Paste")
+                App.ActiveDocument.getObjectsByLabel(
+                    "AeroFoil_" + str(refNum + i + 1).zfill(3)
+                )[0].Label = "AeroFoil_" + str(refNum + i + 1)
+                if self.designprogressbar.value() < 99:
+                    self.designprogressbar.setValue(
+                        round(75 + ((i / (4 * quantity_)) * 100))
+                    )
+        self.designprogressbar.setValue(100)
+        time.sleep(1)
+
+    def characterize(self):
+        try:
+            objRef = App.ActiveDocument.getObjectsByLabel(self.profileName)[0]
+            objRef.addProperty("App::PropertyString", "AirfoilType", "", "", 1)
+            objRef.addProperty("App::PropertyLength", "AirfoilChordLength", "", "", 1)
+            objRef.addProperty("App::PropertyString", "DesignCurveType", "", "", 1)
+            objRef.addProperty("App::PropertyInteger", "NumberOfPoints", "", "", 1)
+            tempStr = ""
+            if self.objRefData[0] == 1:
```

Comment:
All these indexes are very obscure.

Reply:
I've added a comment to explain the labels. It was one of my ideas to lower the line count, by treating some relevant variables as a list.